sdk/search/Azure.Search.Documents/src/SearchServiceClient.cs

```diff
@@ -261,30 +256,309 @@ public virtual SearchIndexClient GetSearchIndexClient(string indexName)
+        /// <returns>The <see cref="Response{T}"/> from the server containing the requested <see cref="DataSource"/>.</returns>
+        /// <exception cref="ArgumentNullException">Thrown when <paramref name="dataSourceName"/> is null.</exception>
+        /// <exception cref="RequestFailedException">Thrown when a failure is returned by the Search service.</exception>
+        [ForwardsClientCalls]
+        public virtual Response<DataSource> GetDataSource(
+            string dataSourceName,
+            SearchRequestOptions options = null,
+            CancellationToken cancellationToken = default) =>
+            DataSourcesClient.Get(
+                dataSourceName,
+                options?.ClientRequestId,
+                cancellationToken);
+
+        /// <summary>
+        /// Gets a specific <see cref="DataSource"/>.
+        /// </summary>
+        /// <param name="dataSourceName">Required. The name of the <see cref="DataSource"/> to get.</param>
+        /// <param name="options">Optional <see cref="SearchRequestOptions"/> to customize the operation's behavior.</param>
+        /// <param name="cancellationToken">Optional <see cref="CancellationToken"/> to propagate notifications that the operation should be canceled.</param>
+        /// <returns>The <see cref="Response{T}"/> from the server containing the requested <see cref="DataSource"/>.</returns>
+        /// <exception cref="ArgumentNullException">Thrown when <paramref name="dataSourceName"/> is null.</exception>
+        /// <exception cref="RequestFailedException">Thrown when a failure is returned by the Search service.</exception>
+        [ForwardsClientCalls]
+        public virtual async Task<Response<DataSource>> GetDataSourceAsync(
+            string dataSourceName,
+            SearchRequestOptions options = null,
+            CancellationToken cancellationToken = default) =>
+            await DataSourcesClient.GetAsync(
+                dataSourceName,
+                options?.ClientRequestId,
+                cancellationToken)
+                .ConfigureAwait(false);
+
+        /// <summary>
+        /// Gets a list of all data sources.
+        /// </summary>
+        /// <param name="select">Optional property names to select. The default is all properties.</param>
+        /// <param name="options">Optional <see cref="SearchRequestOptions"/> to customize the operation's behavior.</param>
+        /// <param name="cancellationToken">Optional <see cref="CancellationToken"/> to propagate notifications that the operation should be canceled.</param>
+        /// <returns>The <see cref="Response{T}"/> from the server containing a list of <see cref="DataSource"/>.</returns>
+        /// <exception cref="RequestFailedException">Thrown when a failure is returned by the Search service.</exception>
+        [ForwardsClientCalls]
+        public virtual Response<IReadOnlyList<DataSource>> GetDataSources(
+            IEnumerable<string> select = null,
```

Comment:
I wonder if we should give these a name like `selectedFields` or something?  Doesn't have to happen as part of this PR.

Reply:
Not a bad idea. The rest would be, based on docs, `selectedProperties`. Seems odd that one would be different. I think I might just standardize on "properties" for consistency, though.

Reply:
I ended up going with `selectProperties` - more a verb form that matches the intent. I could find no solid guideline or preexisting art in our own APIs, but .NET guidelines did mention naming with intent (as oppose to effect). Not married to it, though. Let me know if you would rather have it be `selectedProperties`. I also just considered `properties`, which is fairly consistent in various parts of the .NET BCL, but I figured that's only slightly less ambiguous than `select`; but, I'm also not opposed to that.

Reply:
That sounds good to me.  I think we'll spend some time early next preview looking at the entire surface in ApiView and maybe one of these options will feel like a clear winner.

Reply:
@heaths @tg-msft The main motivation for allowing `$select` for these APIs was to support the scenario where you just list the names of resources without the rest of the objects. So one option would be to expose `GetXYZNames` APIs instead of having this property.